### PR TITLE
Hospital Filters

### PR DESCRIPTION
**Pull request by Sashank999**

### CODE REVIEW

File: extension/scripts/features/hospitalfilter/ttHospitalFilter.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+				const level = parseInt(li.find(".info-wrap .level").innerText.replace(/\D+/g, ""));
+				if ((levelStart && level < levelStart) || (levelEnd !== 100 && level > levelEnd)) {
+					showRow(li, false);
+					continue;
+				}
+				showRow(li);
+			}
+
+			function showRow(li, show = true) {
+				if (show) li.classList && li.classList.remove("hidden");
```
Comment (DeKleineKobini) [inline extension/scripts/features/hospitalfilter/ttHospitalFilter.js:168]: Don't call functions this way.

File: extension/scripts/features/hospitalfilter/ttHospitalFilter.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+				else li.classList && li.classList.add("hidden");
```
Comment (DeKleineKobini) [inline extension/scripts/features/hospitalfilter/ttHospitalFilter.js:169]: Don't call functions this way.

File: extension/scripts/global/functions/slider.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+function newSlider() {
```
Comment (DeKleineKobini) [inline extension/scripts/global/functions/slider.js:1]: This should have a minimum and maximum value, instead of always being 0-100.

**Review by DeKleineKobini — CHANGES_REQUESTED** [review]

File: extension/scripts/features/hospitalfilter/ttHospitalFilter.css [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#hospitalFilter.tt-container {
+	margin-top: 10px;
+}
+
+.statistic {
```
[the remark this answers is not in the document]
Comment (Sashank999) [reply extension/scripts/features/hospitalfilter/ttHospitalFilter.css]: Changed.

File: extension/scripts/global/functions/filters.css [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+.filter-multi-wrap.tt-mobile > * {
+	flex: 1 0 auto;
+}
+
+.tt-checkbox-wrap input {
+    height: 13px;
+    position: relative;
+    top: -2px;
+    cursor: pointer;
+    margin: 1px 5px 0px;
```
[the remark this answers is not in the document]
Comment (Sashank999) [reply extension/scripts/global/functions/filters.css]: Changed.

**Comment by DeKleineKobini** [comment]
1. New sliders seem too 'fat' .
2. Text beneath them isn't centered.
3. They take too much space, so the faction combobox doesn't have enough place
    I'm honestly not a fan of setting a margin to a %, as it mostly looks shit on other sizes. Forcing a width to the combobox isn't needed either imo, a max-width at max.

File: extension/scripts/global/functions/filters.css [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+.filter-slider + .filter-slider-info {
+	width: 100%;
+	text-align: center;
+}
+
+.filter-wrap select {
+	width: 40%;
+	margin-left: 30%;
```
Comment (DeKleineKobini) [inline extension/scripts/global/functions/filters.css:52]: I'm honestly not a fan of setting a margin to a %, as it mostly looks shit on other sizes.

File: extension/scripts/global/functions/filters.css [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+	margin: auto auto 10px;
+}
+
+.filter-slider + .filter-slider-info {
+	width: 100%;
+	text-align: center;
+}
+
+.filter-wrap select {
+	width: 40%;
```
Comment (DeKleineKobini) [inline extension/scripts/global/functions/filters.css:51]: Forcing a width to the combobox isn't needed either imo, a max-width at max.

File: extension/scripts/features/hospitalfilter/ttHospitalFilter.css [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+.tt-mobile #level-filter {
+	max-width: -moz-available; /* WebKit-based browsers will ignore this. */
+	max-width: -webkit-fill-available; /* Mozilla-based browsers will ignore this. */
+	max-width: fill-available;
+	margin-left: 15px;
+}
+
+#hospitalFilter .time-filter-info-wrap,
+#hospitalFilter .level-filter-info-wrap {
+	margin-left: 15%;
```
Comment (DeKleineKobini) [inline extension/scripts/features/hospitalfilter/ttHospitalFilter.css:21]: I'm honestly not a fan of setting a margin to a %, as it mostly looks shit on other sizes.

**Review by DeKleineKobini — CHANGES_REQUESTED** [review]
Comments from above apply.

The sliders aren't centered on mobile as well.

**Review by DeKleineKobini — APPROVED** [review]

**Pull request merged**